src/components/BookList.js

```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
-import Books from './Book';
+import Book from './Book';
 
```

Comment:
Hi, you have a good start with your code, but there are some changes that you need to make to meet the project requirement. You are missing the `useEffect` hook and the `fetchBooks` function in your `BookList` component. The `useEffect` hook is used to perform side effects, such as fetching data from an API, in a React component. The `fetchBooks` function is presumably the action creator that dispatches an action to fetch the books from an API and update the state. You need to import these dependencies and use them in your component. For example:

```js
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Book from './Book';
import { fetchBooks } from '../actions/bookActions';

const BookList = () => {
  const booklist = useSelector((state) => state.book.booklist);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchBooks());
  }, [dispatch]);

  if (!booklist || booklist.length === 0) {
    return <p>No books available.</p>;
  }

  return (
    <>
      {booklist.map((book) => (
        <Book key={book.item_id} item={book} />
      ))}
    </>
  );
};

export default BookList;

Reply:
@topeogunleye  thank you